### DISCUSSION TITLE
How can I ignore one package when updating?

### DISCUSSION THREAD

**Discussion by Hubro** [Morganamilo/paru#813] [Q&A]
I'm trying to update my AUR packages, but the `google-cloud-sdk` package immediately fails to update and stops everything:

```
∷ paru -Su
:: Starting full system upgrade...
 there is nothing to do
:: Looking for AUR upgrades
:: Looking for devel upgrades
:: Resolving dependencies...
:: Calculating conflicts...
:: Calculating inner conflicts...
:: marked out of date: bottles  teamviewer  webkitgtk

Aur (21) anydesk-bin-6.2.0-2  butane-bin-0.15.0-1  celt-0.11.3-5
    expressvpn-3.28.0.6_1-1  google-chrome-103.0.5060.134-1  google-cloud-sdk-394.0.0-1
    icoextract-0.1.3-3  paru-1.11.1-1  postman-bin-9.25.0-1
    proton-ge-custom-bin-1:GE_Proton7_26-1  protontricks-1.9.0-1  qdirstat-1.8.1-1
    signal-desktop-beta-bin-5.51.0beta.1-1  slack-desktop-4.27.156-1  trdsql-bin-0.10.0-1
    ventoy-bin-1.0.78-1  visual-studio-code-bin-1.69.2-1  wowup-2.8.3-1
    neovide-git-0.8.0.r0.g86aa275-1  obs-vkcapture-git-r46.0645e74-1
    solaar-git-1.1.3rc3.r0.g5a3ffe0-1

:: Proceed to review? [Y/n]:

:: Downloading PKGBUILDs...
 (1/1) google-cloud-sdk-394.0.0-1                    [------------------------------------]
error: command failed: /home/tomas/.cache/paru/clone/google-cloud-sdk: git -c user.email=aur -c user.name=aur merge --no-edit --no-ff --no-commit:
```

Is there any way to skip this package, or ignore the error and proceed?

**Comment by kannapoix** [comment]
I think `--ignore` option may works, but as it said in arch wiki, partial update is not supported.
https://wiki.archlinux.org/title/Pacman#Skip_package_from_being_upgraded

**Comment by haplo** [comment]
I'm in this same situation but trying to install `ledger-live-bin`. It depends on `ledger-udev`, which fails for me.

I try `--ignore` but it still tries to install it:

```
$ paru -S --ignore ledger-udev ledger-live-bin
:: Resolving dependencies...
:: Calculating conflicts...
:: Calculating inner conflicts...

:: Conflicts found:
ledger-udev: ledger-udev-git (ledger-udev)

:: Conflicting packages will have to be confirmed manually

Aur (2) ledger-udev-1-8  ledger-live-bin-2.75.0-1

:: Proceed to review? [Y/n]:
```

Same with `--assume-installed`:

```
$ paru -S --assume-installed ledger-udev ledger-live-bin
:: Resolving dependencies...
:: Calculating conflicts...
:: Calculating inner conflicts...

:: Conflicts found:
ledger-udev: ledger-udev-git (ledger-udev)

:: Conflicting packages will have to be confirmed manually

Aur (2) ledger-udev-1-8  ledger-live-bin-2.75.0-1

:: Proceed to review? [Y/n]: n

$ paru -S --assume-installed ledger-udev=1.8 ledger-live-bin
:: Resolving dependencies...
:: Calculating conflicts...
:: Calculating inner conflicts...

:: Conflicts found:
ledger-udev: ledger-udev-git (ledger-udev)

:: Conflicting packages will have to be confirmed manually

Aur (2) ledger-udev-1-8  ledger-live-bin-2.75.0-1

:: Proceed to review? [Y/n]: n
```

My workaround was to edit `ledger-live-bin`'s `PKGBUILD`, drop the `ledger-udev` dependency and then install it with `paru -Bi`. There must be a better way.

**Reply by Morganamilo** [reply]
The assume-installed not working is probably a bug in paru. That way should work.

**Comment by prosoitos** [comment]
I'd like to have such a feature as well but for a different reason:

One of my AUR packages needs to be built and that takes forever. I don't want to do this every time I want to update other AUR packages.

**Reply by prosoitos** [reply]
I can decline to update AUR packages altogether. But I'd like to be able to answer the y/n question for every package.

**Reply by prosoitos** [reply]
I suspect that #1344 asks for exactly that.

**Comment by themarcman314** [comment]
You can add it to the list in the `IgnorePkg` line in your pacman conf `/etc/pacman.conf`

**Reply by prosoitos** [reply]
Oh cool! I didn't realize that packages from the AUR could be added to `pacman.conf` while using `paru`.

Thank you! 🙂

**Reply by prosoitos** [reply]
Just tested it and it works great 🙂 

Thanks again!! I really thought that this was limited to community packages. I guess I hadn't really given much thought to how `paru` works under the hood.

**Reply by themarcman314** [reply]
Yeah I did not know either but there was a similar option on the man and it mentioned this.
I assumed at first it was for paru's config but that didn't work.
I then tried this for pacman's config and it did.
